client/db/src/lib.rs

```diff
@@ -1116,21 +1124,35 @@ impl<Block: BlockT> Backend<Block> {
 				let mut changeset: sc_state_db::ChangeSet<Vec<u8>> = sc_state_db::ChangeSet::default();
 				let mut ops: u64 = 0;
 				let mut bytes: u64 = 0;
+				let mut removal: u64 = 0;
+				let mut bytes_removal: u64 = 0;
 				for (key, (val, rc)) in operation.db_updates.drain() {
 					if rc > 0 {
 						ops += 1;
 						bytes += key.len() as u64 + val.len() as u64;
 
 						changeset.inserted.push((key, val.to_vec()));
 					} else if rc < 0 {
-						ops += 1;
-						bytes += key.len() as u64;
+						removal += 1;
```

Comment:
why removal needs to be counted differently than just writes?

Reply:
Due to the different ways it can be handled in https://github.com/paritytech/substrate/blob/a439a7aa5a9a3df2a42d9b25ea04288d3a0866e8/client/state-db/src/lib.rs#L255
but like I said in https://github.com/paritytech/substrate/pull/4562#discussion_r386535588 we could also use the same counter but only register the ops when we do not run in archive mode; I do not really have a preference: on one hand it is an additional counter, on the other it is a bit implementation related.

Reply:
Yeah thanks, nice